### PR TITLE
samples: matter: Added support for Matter over Wi-Fi in template

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
* Added code necessary to initialize WiFi driver
* Moved OpenThread config options from prj.confs to Kconfig file
* Added nrf7002dk and nrf7002_ek platforms to twister builds
* Extended documentation to include information about Wi-Fi
* Added WiFiDiagnostics cluster support.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
@greg-fer could you take a look at the docs build failure? I have no clue what is the problem with it.

File: samples/matter/template/README.rst [hunk 1]
```diff
@@ -38,9 +47,9 @@ You can press a button to start the factory reset when needed.
 Remote testing in a network
 ===========================
 
-Testing in a Matter-enabled Thread network requires a Matter controller that you can configure either on a PC or a mobile device.
-By default, the Matter accessory device has Thread disabled.
-To use the device within a Thread network, you must pair it with the Matter controller over Bluetooth LE to get the configuration from the controller.
+Testing in a Matter-enabled Thread or Wi-Fi network requires a Matter controller that you can configure either on a PC or a mobile device.
+By default, the Matter accessory device has IPv6 networking disabled.
+ou must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network.
```
Comment (Damian-Nordic) [inline samples/matter/template/README.rst:52]:
```suggestion
You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network.
```

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: samples/matter/template/README.rst [hunk 2]
```diff
@@ -8,8 +8,9 @@ Matter: Template
    :depth: 2
 
 This sample demonstrates a minimal implementation of the :ref:`Matter <ug_matter>` application layer.
-This basic implementation enables the commissioning on the device, which allows it to join a Matter network.
-This sample works as a Thread :ref:`Minimal End Device <thread_ot_device_types>`.
+This basic implementation enables the commissioning on the device, which allows it to join a Matter network built on top of a low-power 802.15.4 Thread or Wi-Fi network.
```
Comment (greg-fer) [inline samples/matter/template/README.rst:11]:
```suggestion
This basic implementation enables the commissioning on the device, which allows it to join a Matter network built on top of a low-power, 802.15.4 Thread network or on top of a Wi-Fi network.
```

File: samples/matter/template/README.rst [hunk 3]
```diff
@@ -38,9 +47,9 @@ You can press a button to start the factory reset when needed.
 Remote testing in a network
 ===========================
 
-Testing in a Matter-enabled Thread network requires a Matter controller that you can configure either on a PC or a mobile device.
-By default, the Matter accessory device has Thread disabled.
-To use the device within a Thread network, you must pair it with the Matter controller over Bluetooth LE to get the configuration from the controller.
+Testing in a Matter-enabled Thread or Wi-Fi network requires a Matter controller that you can configure either on a PC or a mobile device.
```
Comment (greg-fer) [inline samples/matter/template/README.rst:50]:
```suggestion
Testing in either a Matter-enabled Thread or a Wi-Fi network requires a Matter controller that you can configure on PC or mobile device.
```

File: samples/matter/template/README.rst [hunk 4]
```diff
@@ -38,9 +47,9 @@ You can press a button to start the factory reset when needed.
...
+By default, the Matter accessory device has IPv6 networking disabled.
+You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network.
```
Comment (greg-fer) [inline samples/matter/template/README.rst:52]:
```suggestion
You must pair the device with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or a Wi-Fi network.
```

File: samples/matter/template/README.rst [hunk 5]
```diff
@@ -51,8 +60,8 @@ Commissioning in Matter
 In Matter, the commissioning procedure takes place over Bluetooth LE between a Matter accessory device and the Matter controller, where the controller has the commissioner role.
 When the procedure has completed, the device is equipped with all information needed to securely operate in the Matter network.
 
-During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the Thread network credentials to the Matter accessory device.
-As a result, the device can join the Thread network and communicate with other Thread devices in the network.
+During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the Thread or Wi-Fi network credentials to the Matter accessory device.
```
Comment (greg-fer) [inline samples/matter/template/README.rst:63]:
```suggestion
During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the network credentials to the Matter accessory device.
```
Maybe like this?

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by markaj-nordic — APPROVED** [review]

File: samples/matter/template/README.rst (hunk 5 above)
Comment (kkasperczyk-no) [reply samples/matter/template/README.rst:63]: I think it's important to emphasize that those are Thread or Wi-Fi credentials, while "network credentials" may suggest Matter network. I aligned the sentence to the one used by you in generic environment configuration description: https://github.com/nrfconnect/sdk-nrf/pull/8474/files#diff-917767b2147b7f8cdb4953de197f494b42b3fff0ebf0da84cca4853157e66f51R14

**Review by greg-fer — APPROVED** [review]

**Pull request merged**